### PR TITLE
bench: add benchmarks for table name resolution

### LINKED ISSUE #44733: sql: benchmark schema resolution in 19.2 vs 20.1
since we added extra schema resolution steps for temp table, it would be good to double check that this doesn't adversely affect performance in 20.1

### PR DESCRIPTION
**Pull request by otan**
Resolves https://github.com/cockroachdb/cockroach/issues/44733

Release note: None

### CODE REVIEW

File: pkg/bench/bench_test.go [hunk 1]
```diff
@@ -1108,30 +1167,3 @@ func BenchmarkNameResolution(b *testing.B) {
 		b.StopTimer()
 	})
 }
-
-// When temporary tables are present in the system, the PG search path semantics
-// dictate that we try searching for tables under the temporary schema before
-// the public physical schema. This test is used to microbenchmark the effects
-// of this scenario.
-func BenchmarkNameResolutionTempTablesExist(b *testing.B) {
```
Comment (otan) [inline pkg/bench/bench_test.go]: (i think this test is not as useful as the one above)

**Review by solongordon — APPROVED** [review]
Any intuition why resolving `table` is twice as slow as `database.table` when the temp schema exists? Seems to me they should be basically equivalent.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/44828#-:-M-jnbi71ByIyfoPPdjq:bx1ftok)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @solongordon)

___
*[pkg/bench/bench_test.go, line 224 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/44828#-M-jnGk14_U1JHyJYU5G:-M-jnGk14_U1JHyJYU5H:b-gr6gok) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e07461ac1fe48e460cdb60cfb970a01e104e400/pkg/bench/bench_test.go#L224)):*
> ```Go
> 
> 	for _, createTempTables := range []bool{false, true} {
> 		b.Run(fmt.Sprintf("create_temp_tables:%t", createTempTables), func(b *testing.B) {
> ```

Maybe clearer to call it `tempSchemaExists` instead of `create_temp_tables`.

**Comment by otan** [comment]
> Any intuition why resolving table is twice as slow as database.table when the temp schema exists? Seems to me they should be basically equivalent.

Hmm, nope. I'll try dig into this in at a later time!

**Comment by otan** [comment]
Found out - because searching for `database.table` only does `defaultdb.defaultdb.b` and `defaultdb.public.t` (ignoring temp, as we do not allow `D.T` to implicitly recall non-public schemas), whereas searching for just `table` searches `defaultdb.pg_temp.b` and `defaultdb.b`

**Comment by otan** [comment]
bors r=solongordon

**Comment by solongordon** [comment]
Got it. That's slightly confusing but I guess it's consistent with how we treat virtual schemas like `pg_catalog` so seems fine. (And of course there is no Postgres behavior to mimic in this case.)

**Pull request merged**